cupy/core/_routines_statistics.pyx

```diff
@@ -310,6 +311,57 @@ cdef _nanargmax_func = create_reduction_func(
     None, _min_max_preamble)
 
 
+cpdef ndarray _median(
+        ndarray a, axis, out, overwrite_input, keepdims):
+
+    if not isinstance(a, ndarray):
+        raise TypeError('Array is not cupy.ndarray')
+
+    keep_ndim = a.ndim
+
+    if axis is None:
+        sz = a.size
+    else:
+        sz = a.shape[axis]
+    if sz % 2 == 0:
+        szh = sz // 2
+        kth = [szh - 1, szh]
+    else:
+        kth = [(sz - 1) // 2]
+
+    if overwrite_input:
+        part = a
+    else:
+        part = a.copy()
+
+    if axis is None:
+        part = part.ravel()
+        part.partition(kth)
+    else:
+        part.partition(kth, axis=axis)
+
+    if part.shape == ():
+        return part.item()
```

Comment:
I think this case should just return a 0-dimensional array for consistency with other CuPy functions like `mean`? item() will cause host->device transfer.

Reply:
good catch